src/Components/Welcome/index.js

```diff
@@ -1,74 +1,60 @@
+  user(id: 1) {
+    firstName lastName email roles { name }
   }
 }`;
 
 @QueryHOC(query)
+@withRouter
 export default class Navbar extends Component {
-    render() {
-        const { loading, error, data: {user} } = this.props.queryResults;
-        if (loading) return "Loading graphql query..";
-        if (error) return `GraphQL query resulted in error: ${error}`;
-        return (
-            <div className="welcome">
-                <tr>
-                    <td>
-                        Voornaam:
-                    </td>
-                    <td>
-                        &nbsp;&nbsp;
-                    </td>
-                    <td>
-                        {user.firstName}
-                    </td>
-                </tr>
-                <tr><br/></tr>
-                <tr>
-                    <td>
-                        Achternaam:
-                    </td>
-                    <td>
-                        &nbsp;&nbsp;
-                    </td>
-                    <td>
-                        {user.lastName}
-                    </td>
-                </tr>
-                <tr><br/></tr>
-                <tr>
-                    <td>
-                        Email:
-                    </td>
-                    <td>
-                        &nbsp;&nbsp;
-                    </td>
-                    <td>
-                        {user.email}
-                    </td>
-                </tr>
-                <tr><br/></tr>
-                <tr>
-                    <td>
-                        Rol:
-                    </td>
-                    <td>
-                        &nbsp;&nbsp;
-                    </td>
-                    <td>
-                        {user.roles.map(userToJSX)}
-                    </td>
-                </tr>
-            </div>
-        );
-    }
+  logout = this.logout.bind(this);
+
+  logout() {
+    localStorage.clear();
+    this.props.history.push("/login");
+  }
+
+  render() {
+    const { loading, error, data} = this.props.queryResults;
+    if (loading) return "Loading graphql query..";
+    if (error) return `GraphQL query resulted in ${error}`;
+    const { user } = data;
+
+    console.log("BasicAuth = ", atob(localStorage.getItem("BasicAuth")), localStorage.getItem("BasicAuth"));
+    return (
+      <div className="welcome">
+        <table>
+          <tbody>
+            <tr>
+              <td> Voornaam: </td>
+              <td> {user.firstName} </td>
+            </tr>
+            <tr>
+              <td> Achternaam: </td>
+              <td> {user.lastName} </td>
+            </tr>
+            <tr>
+              <td> Email: </td>
+              <td> {user.email} </td>
+            </tr>
+            <tr>
+              <td> Rollen: </td>
+              <td><ul>{user.roles.map(userToJSX)}</ul></td>
+            </tr>
+          </tbody>
+        </table>
+          <input type="submit" className="logout" value="Logout" onClick={this.logout} />
```

Comment:
```suggestion
        <input type="submit" className="logout" value="Logout" onClick={this.logout} />
```

Reply:
We gaan nog even discussieren over die 2 space vs 4 space tabs